gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterStatsMonitor.java

```diff
@@ -27,4 +31,12 @@ static ClusterStats.Builder getClusterStatsBuilder(ProxyBackendConfiguration bac
         builder.routingGroup(backend.getRoutingGroup());
         return builder;
     }
+
+    static boolean shouldRetry(int statusCode)
```

Comment:
This method is related to only `INFO_API` and `METRICS`. I would extract into helper static class instead of interface.